### PR TITLE
SI-6616 Check that unsafe operations are only called on the presentation...

### PR DESCRIPTION
**Pull request by dragos**
... compiler thread.

The method that checks the actual constraint is @elidable, expecting it to be used
for nightly builds but stripped-off in release builds. This way we don't lose any
performance, but 'fail-fast' in IDE nightlies.

This assumes that release builds will have at least `-Xelide-below ASSERTION`, but
this pull request does not do that.

review by @adriaanm, @paulp

### CODE REVIEW

**Comment by jsuereth** [comment]
Yet another initialization tracker....  @paulp didn't you add another one of these recently in typer?

**Comment by paulp** [comment]
Yes, that is either redundant with 'isGlobalInitialized', or we should make it so. (By which I mean, if isGlobalInitialized doesn't have exactly the desired semantics for this, I am sure we can find something which satisfies all uses.)

**Comment by dragos** [comment]
Thanks for pointing that out. My guess is that I can use the other method, I only need to eliminate some false positives. Even if it's not as precise as my variable, it would still be ok. I'll make the changes ASAP.

**Comment by dragos** [comment]
..the only problem being that `isGlobalInitialized` is not in 2.10.x, only in 2.11. Any thoughts on that?

**Comment by jsuereth** [comment]
@paulp is there any reason not to cherry-pick isGlobalInitialzied from master -> 2.10.x?

**Comment by paulp** [comment]
No reason.

**Comment by dragos** [comment]
I'll close this pull request and open another one with the cherry-pick.

**Comment by dragos** [comment]
The commit itself is small, but depends on a bunch of other refactorings and can't be easily backported to 2.10.x anymore. I decided to keep this pull request for 2.10.x, and open a new one for 2.11 using the new support.

**Comment by dragos** [comment]
So guys, WDYT?

**Comment by paulp** [comment]
Looks good to me.

File: src/compiler/scala/tools/nsc/interactive/Global.scala [hunk 1]
```diff
@@ -51,6 +56,7 @@ class Global(settings: Settings, _reporter: Reporter, projectName: String = "")
   import log.logreplay
   debugLog("logger: " + log.getClass + " writing to " + (new java.io.File(logName)).getAbsolutePath)
   debugLog("classpath: "+classPath)
+  Console.err.println("\n ======= CHECK THREAD ACCESS compiler build ========\n")
```
Comment (adriaanm) [inline src/compiler/scala/tools/nsc/interactive/Global.scala:59]: oops -- how come all reviewers missed this? :-)

Comment (paulp) [inline src/compiler/scala/tools/nsc/interactive/Global.scala:59]: Clearly the reviewers in this case were a bunch of scrubs.

Comment (dragos) [inline src/compiler/scala/tools/nsc/interactive/Global.scala:59]: It's only printing that in the presentation compiler, if that's any consolation :)

**Pull request merged**